src/credentials/auth.ts

```diff
@@ -1199,6 +1202,28 @@ const addConnection = Commands.register('aws.auth.addConnection', async () => {
     }
 })
 
+export async function tryAddCredentials(
+    profileName: SectionName,
+    profileData: StaticCredentialsProfileData,
+    tryConnect = true
+): Promise<boolean> {
+    if (getStaticCredentialsDataErrors(profileData) || (await profileExists(profileName))) {
+        return false
+    }
+
+    await saveProfileToCredentials(profileName, profileData)
+    if (tryConnect) {
+        const auth = Auth.instance
+        const conn = await auth.getConnection({ id: profileName })
+        if (conn === undefined) {
+            throw new ToolkitError('Failed to get connection from profile', { code: 'MissingConnection' })
+        }
```

Comment:
I'm not sure I understand, this code is retrieving and using the connection and not creating it?

Reply:
```ts
await saveProfileToCredentials(profileName, profileData)
```

This line is "creating" a connection. Otherwise how would you get a connection that doesn't exist? :)